soundata/datasets/starss2022.py

```diff
@@ -0,0 +1,645 @@
+    """Validate if STARSS 2022 timesteps are well-formed.
+    If locations is None, validation passes automatically
+    Args:
+        time_step (float): spacing between location steps
+        locations (np.ndarray): (n x 3) array
+        interval (np.ndarray): (n x 2) expected start and end time
+            for the locations
+    Raises:
+        ValueError: if the number of locations does not match
+            the number of time_steps that fit in the interval
+    """
+    if interval[0] > interval[1]:
+        raise ValueError("The interval has a start_time greater than the end_time")
+    elif len(locations) == 1 and interval[1] - interval[0] > 0:
+        pass  # the event is static
+    # if the object is static, validation passes
+    elif not np.isclose(len(locations) - 1, (interval[1] - interval[0]) / time_step):
+        raise ValueError(
+            "The number of locations does not fit in the interval, given the time_step"
+        )
+
+
+def validate_locations(locations):
+    """Validate if STARSS 2022 locations are well-formed.
+    If locations is None, validation passes automatically
+    Args:
+        locations (np.ndarray): (n x 3) array
+    Raises:
+        ValueError: if locations have an invalid shape or
+                have cartesian coordinate values outside the expected ranges.
+    """
+
+    # validate that locations have the correct shape
+    locations_shape = np.shape(locations)
+    if len(locations_shape) != 2 or locations_shape[1] != 3:
+        raise ValueError(
+            f"Locations should be arrays with three columns, but array has shape {locations_shape}"
+        )
+
+    # validate that values are within expected ranges
+    if (np.abs(locations[:, 0]) > 90).any():
+        raise ValueError(f"Elevation values should have magnitude less than 90")
+    if (np.abs(locations[:, 1]) > 180).any():
+        raise ValueError(f"Azimuth values should have magnitude less than 180")
+    elif (locations[:, 2] != None).any():
+        raise ValueError(f"Distance values should be nonnegative numbers")
+
+
+class Clip(core.Clip):
+    """STARSS 2022 Track class
+    Args:
+        clip_id (str): id of the clip
+    Attributes:
+        audio_path (str): path to the audio file
+        tags (soundata.annotation.Tags): tag
+        track_id (str): track id
+        spatial_events (SpatialEvents): sound events with time step, elevation, azimuth, distance, label, track_number and confidence.
+    """
+
+    def __init__(self, clip_id, data_home, dataset_name, index, metadata):
+
+        super().__init__(
+            clip_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+
+        self.audio_path = self.get_path("audio")
+        self.csv_path = self.get_path("events")
+        self.format = self._clip_metadata.get("format")
+        self.set = self._clip_metadata.get("set")
+        self.split = self._clip_metadata.get("split")
```

Comment:
Should `format`, `set` and `split` be defined as a property, like in other loaders. In any case, those are not defined in the docstring of the class.

Reply:
We did not do this for 

TAU SSE 2019
TAU NIGENS SSE 2020
TAU NIGENS SSE 2021

All of which share the same metadata format.

Reply:
I see. But please update the docstring of the Clip class. The attributes are not well documented.

Reply:
Thank you for the clarification. I see what you mean. I have updated the docstring.